### PR TITLE
avocado.core.test: Avoid passing ugly exceptions

### PR DESCRIPTION
**Pull request by ldoktor**
Currently when one passes exception, which is not instance of Exception,
avocado proceeds and returns "INTERRUPTED" test instead of "ERROR". This
for example happens when old-style-class is used as exception. This
patch uses pure "except" without any argument to catch such exceptions
and then it wraps them in "exceptions.TestError" instead.

The same applies to "setUp" and "tearDown", they only use different
class as a wrapper.

Signed-off-by: Lukáš Doktor <ldoktor@redhat.com>

### CODE REVIEW

File: avocado/core/test.py [hunk 1]
```diff
@@ -359,8 +359,9 @@ def _run_avocado(self):
         except exceptions.TestNAError, details:
             stacktrace.log_exc_info(sys.exc_info(), logger='avocado.test')
             raise exceptions.TestNAError(details)
-        except Exception, details:
+        except:     # Old-style exceptions are not inherited from Exception()
             stacktrace.log_exc_info(sys.exc_info(), logger='avocado.test')
+            details = sys.exc_info()[1]
```
Comment (clebergnu) [inline avocado/core/test.py]: "Loud" review:

```
Help on built-in function exc_info in module sys:

exc_info(...)
    exc_info() -> (type, value, traceback)
```

Check!

File: avocado/core/test.py [hunk 2]
```diff
@@ -359,8 +359,9 @@ def _run_avocado(self):
         except exceptions.TestNAError, details:
             stacktrace.log_exc_info(sys.exc_info(), logger='avocado.test')
             raise exceptions.TestNAError(details)
-        except Exception, details:
+        except:     # Old-style exceptions are not inherited from Exception()
```
Comment (clebergnu) [inline avocado/core/test.py]: OK.

File: avocado/core/test.py [hunk 3]
```diff
@@ -371,8 +372,11 @@ def _run_avocado(self):
                                 'must fix your test. Original skip exception: '
                                 '%s' % details)
             raise exceptions.TestError(skip_illegal_msg)
-        except Exception, details:
+        except:     # Old-style exceptions are not inherited from Exception()
             stacktrace.log_exc_info(sys.exc_info(), logger='avocado.test')
+            details = sys.exc_info()[1]
+            if not isinstance(details, Exception):  # Avoid passing nasty exc
+                details = exceptions.TestError("%r: %s" % (details, details))
```
Comment (clebergnu) [inline avocado/core/test.py]: OK, `TestError` is indeed appropriate here IMHO.

File: avocado/core/test.py [hunk 4]
```diff
@@ -384,9 +388,10 @@ def _run_avocado(self):
                                     'you must fix your test. Original skip '
                                     'exception: %s' % details)
                 raise exceptions.TestError(skip_illegal_msg)
-            except Exception, details:
+            except:     # avoid old-style exception failures
                 stacktrace.log_exc_info(sys.exc_info(), logger='avocado.test')
-                cleanup_exception = details
+                details = sys.exc_info()[1]
+                cleanup_exception = exceptions.TestSetupFail(details)
```
Comment (clebergnu) [inline avocado/core/test.py]: Also OK for `TestSetupFail`.

File: examples/tests/failtest_nasty3.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+class FailTest(Test):
+
+    """
+    This test raises old-style-class exception
+    """
+
+    def test(self):
+        """
+        Should fail not-that-badly
```
Comment (clebergnu) [inline examples/tests/failtest_nasty3.py]: Maybe a docstring like this: "Test is broken and Avocado should handle it by reporting a test ERROR."

File: selftests/functional/test_standalone.py [hunk 6]
```diff
@@ -62,6 +62,13 @@ def test_failtest_nasty2(self):
         self.assertIn("Exception: Unable to get exception, check the traceback"
                       " for details.", result.stdout)
 
+    def test_failtest_nasty3(self):
```
Comment (clebergnu) [inline selftests/functional/test_standalone.py]: The name of the test is misleading, because we want to get a `TestError`. Maybe it should be called `errortest_nasty.py` ?

Comment (ldoktor) [inline selftests/functional/test_standalone.py]: Well, that's true. Anyway as there are 3 of them, I'd prefer to do it as separate PR...
